### PR TITLE
add additional test cases for service visibility

### PR DESCRIPTION
**Pull request by ramaraochavali**
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/model/push_context_test.go [hunk 1]
```diff
@@ -859,6 +859,47 @@ func TestIsServiceVisible(t *testing.T) {
...
+		},
+		{
+			name:        "service has both public visibility and none visibility",
+			pushContext: &PushContext{},
+			service: &Service{
+				Attributes: ServiceAttributes{
+					Namespace: "bar",
+					ExportTo: map[visibility.Instance]bool{
+						visibility.Public: true,
+						visibility.None:   true,
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/push_context_test.go:883]: I think this is not a right config, maybe we should prevent it ealier

File: pilot/pkg/model/push_context_test.go [hunk 2]
```diff
@@ -859,6 +859,47 @@ func TestIsServiceVisible(t *testing.T) {
...
+		},
+		{
+			name:        "service has both none visibility and private visibility",
+			pushContext: &PushContext{},
+			service: &Service{
+				Attributes: ServiceAttributes{
+					Namespace: "bar",
+					ExportTo: map[visibility.Instance]bool{
+						visibility.Private: true,
+						visibility.None:    true,
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/push_context_test.go:897]: same here not invalid case

File: pilot/pkg/model/push_context_test.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/model/push_context_test.go:883]: Current code supports it and the comments are very clear on what to do when this is configured. That is why I added these tests to ensure the order is not messed up in future. I do not know if we can prevent this/change this now for BC reasons.

**Review by hzxuzhonghu — APPROVED** [review]
keep the previous behavior

**Pull request merged**